apps/server/src/modules/tool/service/external-tool-version.service.ts

```diff
@@ -0,0 +1,62 @@
+import { Injectable } from '@nestjs/common';
+import { CustomParameterDO, ExternalToolDO } from '@shared/domain/domainobject/external-tool';
+
+@Injectable()
+export class ExternalToolVersionService {
+	increaseVersionOfNewToolIfNecessary(oldTool: ExternalToolDO, newTool: ExternalToolDO): void {
+		if (!oldTool.parameters || !newTool.parameters) {
+			return;
+		}
+		if (this.compareParameters(oldTool.parameters, newTool.parameters)) {
+			newTool.version += 1;
+		}
+	}
+
+	private compareParameters(oldParams: CustomParameterDO[], newParams: CustomParameterDO[]): boolean {
+		const matchingParams: CustomParameterDO[] = oldParams.filter((oldParam) => {
+			return newParams.some((newParam) => oldParam.name === newParam.name);
+		});
+
+		const shouldIncrementVersion =
+			this.hasNewRequiredParameter(oldParams, newParams) ||
+			this.hasChangedRequiredParameters(oldParams, newParams) ||
+			this.hasChangedParameterNames(oldParams, newParams) ||
+			this.hasChangedParameterRegex(newParams, matchingParams) ||
+			this.hasChangedParameterTypes(newParams, matchingParams);
+
+		return shouldIncrementVersion;
+	}
+
+	private hasNewRequiredParameter(oldParams: CustomParameterDO[], newParams: CustomParameterDO[]): boolean {
+		return newParams.some(
+			(newParam) => !newParam.isOptional && oldParams.every((oldParam) => oldParam.name !== newParam.name)
+		);
+	}
+
+	private hasChangedParameterNames(oldParams: CustomParameterDO[], newParams: CustomParameterDO[]): boolean {
+		const names1 = oldParams.map((parameter) => parameter.name);
+		const names2 = newParams.map((parameter) => parameter.name);
+		return !!(names1.some((name) => !names2.includes(name)) || names2.some((name) => !names1.includes(name)));
```

Comment:
Can you please avoid directly returns (on all places) please first put it to a named const xxx = ... with return xxx;